cars/app/cars/models.py

```diff
@@ -7,16 +8,36 @@ class Customer(models.Model):
 
     def buy_car(self, car_id):
         car = Car.objects.get(id=car_id)
-        car.customer = self;
-        car.save()
+        self.money -= car.price
```

Comment:
Use separate if-statement for the balance check. 
Raise if car.price is higher than the current balance value.

if something:
   raise something 
____

One important thing: use it in signal, not here